app/uk/gov/hmrc/helptosavefrontend/controllers/HelpToSaveAuth.scala

```diff
@@ -63,6 +63,29 @@ class HelpToSaveAuth(frontendAuthConnector: FrontendAuthConnector, metrics: Metr
         }
     }
 
+  def authorisedForHtsWithNINOAndName(action: HtsAction[HtsContextWithNINOAndName])(redirectOnLoginURL: String): Action[AnyContent] =
+    Action.async { implicit request ⇒
+      val timer = metrics.authTimer.time()
+
+      authorised(AuthWithCL200)
+        .retrieve(Retrievals.name and Retrievals.itmpName and authorisedEnrolments) {
+          case maybeName ~ maybeItmpName ~ authorisedEnrols ⇒
+            val time = timer.stop()
```

Comment:
what happens to the `timer count` if its reaches `recover` block ?

Reply:
Yeah you're right - we need to stop it in the recover block